config/config.go

```diff
@@ -0,0 +1,110 @@
+package config
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/knq/ini"
+	"github.com/knq/ini/parser"
+)
+
+type KV struct {
+	Key   string
+	Value string
+}
+
+type Generator struct {
+	Command string
+	Params  []KV
+}
+
+type Config struct {
+	Dir        string
+	Patterns   []string
+	Generators []Generator
+}
+
+func Load(path string) (*Config, error) {
+	origPath := path
+	var err error
+	if path == "" {
+		path, err = os.Getwd()
+		if err != nil {
+			return nil, fmt.Errorf("error getting working directory: %v", err)
+		}
+	}
+
+	var cfg *Config
+	for {
+		configPath := filepath.Join(path, ".gunkconfig")
+		reader, err := os.Open(configPath)
+		if err != nil {
+			path = filepath.Dir(path)
+			if path == "/" {
+				return nil, fmt.Errorf("could not find a .gunkconfig")
+			}
+			continue
+		}
+		defer reader.Close()
+		cfg, err = load(reader)
+		if err != nil {
+			return nil, fmt.Errorf("error loading %q: %v", configPath, err)
+		}
+		if cfg != nil {
+			cfg.Dir = origPath
+			break
+		}
+	}
+
+	return cfg, nil
+}
+
+func load(reader io.Reader) (*Config, error) {
+	f, err := ini.Load(reader)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse ini file: %v", err)
+	}
+	config := &Config{
+		Generators: make([]Generator, 0, len(f.AllSections())),
+	}
+	for _, s := range f.AllSections() {
+		var err error
+		var gen *Generator
+		switch s.Name() {
+		case "":
+			// This seems to happen sometimes?
+		case "generate":
+			gen, err = handleGenerate(s)
+		default:
+			return nil, fmt.Errorf("unknown section %q", s.Name())
+		}
+		if err != nil {
+			return nil, err
+		}
+		if gen != nil {
+			config.Generators = append(config.Generators, *gen)
+		}
+
+		gen = nil
```

Comment:
I'm pretty sure this line is unnecessary, since you declare the pointer variable at every iteration